lib/spack/spack/test/data/compilers.yaml

```diff
@@ -114,3 +114,12 @@ compilers:
       cflags: -O3
       cxxflags: -O3
     modules: 'None'
+- compiler:
```

Comment:
I looked into this and you should not have to define this. At the moment it appears Spack is being a bit loose and using compilers from other specs with different architectures. I'm looking into fixing this. Perhaps I could add that as a commit to this PR, although it may make more sense to open a new PR on it.

Reply:
I got rid of it but it is causing a concretize test to fail. I'm guessing this PR will not be merged until the compiler concretization is fixed?